### PR TITLE
Update release notes, process, platforms.

### PR DESCRIPTION
**Pull request by btm**
Platforms have changed significantly since the last release.
So has the release process.

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,5 +1,43 @@
...
+Dependency updates:
+- Ruby 2.6.6
+- Chef Infra Client 15.10.12
+- ohai 15.9.1
+- bzip2 1.0.8
+- cacerts 2019-10-16
+- libarchive 3.4.2
+- libxml2 2.9.10
+- libxslt 1.1.34
+- openssl 1.0.2u
```
Comment (tas50) [inline RELEASE_NOTES.md:16]: This should probably get split out and the security updates noted

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -1,5 +1,43 @@
 # Push Jobs Client Release Notes
 
+## 3.0
+
+Versions of Chef Push Jobs 3.0 and later may only be used under the terms of the [Chef EULA](https://www.chef.io/end-user-license-agreement/) or another commercial agreement with Chef Software.
+
+Dependency updates:
+- Ruby 2.6.6
```
Comment (tas50) [inline RELEASE_NOTES.md:8]: The CVEs fixed here should get noted. It's all in the chef infra client release notes if you need the links

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -1,5 +1,43 @@
...
+- Chef Infra Client 15.10.12
+- ohai 15.9.1
+- bzip2 1.0.8
```
Comment (tas50) [inline RELEASE_NOTES.md:11]: this misc stuff probably isn't worth noting. I doubt anyone cares about ohai, bzip2, cacerts etc

File: RELEASE_NOTES.md [hunk 4]
```diff
@@ -1,5 +1,43 @@
...
+- libxml2 2.9.10
+- libxslt 1.1.34
+- openssl 1.0.2u
+
+Added platform support:
+- AIX 7.2
+- Debian 9, 10
+- RHEL 8
+- macOS 15
+- Ubuntu 16.04, 18.04, 20.04
```
Comment (tas50) [inline RELEASE_NOTES.md:23]: Did we really only support 14.04 before?

File: RELEASE_NOTES.md [hunk 5]
```diff
@@ -1,5 +1,43 @@
...
+- Windows Server 2016, 2019
```
Comment (tas50) [inline RELEASE_NOTES.md:24]: 10 too right?

File: RELEASE_NOTES.md [hunk 6]
```diff
@@ -1,5 +1,43 @@
...
+
+Removed platform support:
```
Comment (tas50) [inline RELEASE_NOTES.md:26]:
```suggestion
Removed EOL platform support:
```

File: RELEASE_NOTES.md (hunk 4 above)
Comment (btm) [reply RELEASE_NOTES.md:23]: no. also 12.04 and 10.04 apparently. heh. we didn't have a `.expeditor/release.omnibus.yml` for the last release,

File: RELEASE_NOTES.md (hunk 2 above)
Comment (btm) [reply RELEASE_NOTES.md:8]: it's going to be a really long list based on how far these have bumped.

Comment (btm) [reply RELEASE_NOTES.md:8]: @tas50 do you know of a good source for the lists? https://www.cvedetails.com/vendor/12872/Libarchive.html seems wrong.

Comment (btm) [reply RELEASE_NOTES.md:8]: yeah the list of CVEs between Ruby 2.4.5 and 2.6.6 are too confusing. I'm just gonna say CVEs fixed.

File: RELEASE_NOTES.md [hunk 7]
```diff
@@ -1,5 +1,45 @@
...
+- libarchive 3.3.3 to 3.4.2
+- libxml2 2.9.7 to 2.9.10
+- libxslt 1.1.30 to 1.1.34
+- openssl 1.0.2p to 1.0.2u
+
+Added platform support:
+- AIX 7.2
+- Debian 9, 10
+- RHEL 8
+- macOS 15
```
Comment (tas50) [inline RELEASE_NOTES.md:24]:
```suggestion
- macOS 10.15
```

File: RELEASE_NOTES.md [hunk 8]
```diff
@@ -1,5 +1,45 @@
...
+- Ubuntu 16.04, 18.04, 20.04
+- Windows Server 2016, 2019, 10
+
+Removed EOL platform support:
+- Debian 6, 7
+- macOS 12
```
Comment (tas50) [inline RELEASE_NOTES.md:30]:
```suggestion
- macOS 10.12
```

**Pull request merged**